compiler/typecheck/TcDaml.hs

```diff
@@ -41,10 +42,10 @@ import qualified Prelude as P ((<>))
 import Control.Monad
 import Data.Foldable (fold)
 
-check :: NamedThing a => [String] -> String -> a -> Bool
-check modules name namedThing
-  | Just mod <- moduleNameString . moduleName <$> nameModule_maybe (getName namedThing)
-  , mod `elem` modules
+check :: NamedThing a => a -> (String, String) -> Bool
+check namedThing (mod, name)
+  | Just mod' <- moduleNameString . moduleName <$> nameModule_maybe (getName namedThing)
+  , mod == mod'
```

Comment:
Maybe a wrapper thats simply
```hs
checkAny :: NamedThing a => a -> [String] -> [String] -> Bool
checkAny namedThing mods names = any (check namedThing) $ liftA2 (,) mods names
```
Which you can reuse for all the cases below and reduce clutter?

Reply:
yeah it did get a bit cluttered. I've done something like your suggestion, except as part of `check` itself